Userland/Libraries/LibWasm/AbstractMachine/AbstractMachine.h

```diff
@@ -79,41 +79,54 @@ class Value {
     {
     }
 
-    using AnyValueType = Variant<i32, i64, float, double, u128, Reference>;
-    explicit Value(AnyValueType value)
-        : m_value(move(value))
+    template<typename T>
+    requires(sizeof(T) == sizeof(u64)) explicit Value(T raw_value)
+        : m_value(u128(bit_cast<i64>(raw_value), 0))
+    {
+    }
+
+    template<typename T>
+    requires(sizeof(T) == sizeof(u32)) explicit Value(T raw_value)
+        : m_value(u128(static_cast<i64>(bit_cast<i32>(raw_value)), 0))
     {
     }
 
     template<typename T>
-    requires(sizeof(T) == sizeof(u64)) explicit Value(ValueType type, T raw_value)
+    requires(sizeof(T) == sizeof(u8) && Signed<T>) explicit Value(T raw_value)
```

Comment:
I'm certain that there's better template logic that we can do here...

Reply:
Eh, looks alright.